src/main/webapp/WEB-INF/apsadmin/jsp/common/tiles-inserts/header.jsp

```diff
@@ -17,9 +17,19 @@
 <script>
     jQuery(function () {
         $('#activity-stream [data-toggle="tooltip"]').tooltip({trigger: 'hover'});
+
+        // trigger updateLocale if language is changed
+        $('#languageDropdownComponent a').click(function(e) {
+            e.preventDefault();
+            if (updateLocalStorageWithLocale) {
+                updateLocalStorageWithLocale($(this).data('locale'));
+            }
+            window.location.href = $(this).attr('href');
```

Comment:
is this always needed?

Reply:
we need to intercept the link that changes the language, and call `updateLocalStorageWithLocale` to sync the localStorage with the new language, so if users changes to AP, the new language will be already set.

`updateLocalStorageWithLocale` will only be declared in `menuAppBuilderIntegration.jsp` which is flag-protected, so for other cases, the line 24 will be false and the localStorage won't be updated.